chore: add info to the new and noteworthy section (#1248)
vaadin/hilla: scripts/generator/templates/template-release-note.md

## New and Noteworthy since Previous Hilla Update

> Hilla 2.2 introduces an innovative form hook for React with server-side validation, and improved hot reloading for faster development.

Reviewer: ```suggestion
Hilla 2.2 introduces a form hook for React with server-side validation, and improved hot reloading for faster development.
```
